service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/dataset/CopyBigQueryDatasetDifferentRegionStep.java

```diff
@@ -0,0 +1,155 @@
+
+    TransferConfig transferConfig =
+        TransferConfig.newBuilder()
+            .setDestinationDatasetId(destinationDatasetId)
+            .setDisplayName(destinationDatasetId)
+            .setScheduleOptions(ScheduleOptions.newBuilder().setDisableAutoScheduling(true))
+            .setDataSourceId("cross_region_copy")
+            .setParams(Struct.newBuilder().putAllFields(params).build())
+            .build();
+
+    try (DataTransferServiceClient dataTransferServiceClient = DataTransferServiceClient.create()) {
+      ProjectName parent = ProjectName.of(destinationProjectId);
+      String petSaEmail =
+          SamRethrow.onInterrupted(
+              () ->
+                  samService.getOrCreatePetSaEmail(
+                      gcpCloudContextService.getRequiredGcpProject(destinationWorkspaceId),
+                      userRequest.getRequiredToken()),
+              "enablePet");
+
+      TransferConfig config =
+          dataTransferServiceClient.createTransferConfig(
+              CreateTransferConfigRequest.newBuilder()
+                  .setParent(parent.toString())
+                  .setTransferConfig(transferConfig)
+                  .setServiceAccountName(petSaEmail)
+                  .build());
+
+      var now = Instant.now();
+
+      dataTransferServiceClient.startManualTransferRuns(
+          StartManualTransferRunsRequest.newBuilder()
+              .setParent(config.getName())
+              .setRequestedRunTime(
+                  com.google.protobuf.Timestamp.newBuilder()
+                      .setSeconds(now.getEpochSecond())
+                      .setNanos(now.getNano()))
+              .build());
+
+      DataTransferServiceClient.ListTransferRunsPagedResponse runs =
+          dataTransferServiceClient.listTransferRuns(config.getName());
+
+      for (TransferRun run : runs.iterateAll()) {
+        final String currentRunName = run.getName();
+        // Wait for job to complete
+        int sleepTimeSeconds = 1;
+        while (true) {
+          final TransferRun currentRun = dataTransferServiceClient.getTransferRun(currentRunName);
+          final TransferState runState = currentRun.getState();
+
+          logger.debug("Run {} is {}", currentRunName, runState);
+          // Data transfer run has finished. It is either SUCCEEDED (4), FAILED (5), or CANCELLED
+          // (6)
+          if (currentRun.getStateValue() >= 4) {
+            if (!"SUCCEEDED".equals(runState.toString())) {
+              final String errorMessage = currentRun.getErrorStatus().getMessage();
+              logger.warn("Job {} failed: {}", currentRunName, errorMessage);
+              return new StepResult(
+                  StepStatus.STEP_RESULT_FAILURE_FATAL, new RuntimeException(errorMessage));
+            }
+            break;
+          }
+          TimeUnit.SECONDS.sleep(sleepTimeSeconds);
+          sleepTimeSeconds = Math.min(2 * sleepTimeSeconds, 60);
```

Comment:
can you extract this retry similar to https://github.com/DataBiosphere/terra-workspace-manager/blob/5829c1db8cc1bcfe22e87f8363fed575c11ac41b/service/src/main/java/bio/terra/workspace/common/utils/RetryUtils.java#L38 in RetryUtil. Maybe something like `getWithRetry` and pass in a precondition as a param. So we retry if precondition fails instead of on exception.

Reply:
Ok, made a function in `RetryUtils` named  `getWithRetry`  that takes in a predicate. If the predicate evaluates false, then we retry.